### PR TITLE
Core/Spells: do not save Liquid auras

### PR DESCRIPTION
**Pull request by Gooyeth**
**Changes proposed:**

Avoid if a pet dies by magma, when invoked always die for magma.

**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [x] master

**Tests performed:** Build and tested ingame

### CODE REVIEW

**Comment by tkrokli** [comment]
Replaces PR #21682

**Comment by tkrokli** [comment]
@ariel- Do you find this solution acceptable & practical for TC?

**Comment by ariel-** [comment]
There must be some generic way. There are other liquid auras being saved besides this one. So the logic would be, any aura that depends on environment shouldn't be saved (ie area auras, liquid auras)

**Comment by Gooyeth** [comment]
Ready, thanks.

Tested ingame 👍

File: src/server/game/Spells/Auras/SpellAuras.cpp [hunk 1]
```diff
@@ -1161,6 +1160,11 @@ bool Aura::CanBeSaved() const
     if (GetCastItemGUID() && IsPermanent())
         return false;
 
+    // don't save liquid auras
+    for (LiquidTypeEntry const* liquid : sLiquidTypeStore)
```
Comment (Shauren) [inline src/server/game/Spells/Auras/SpellAuras.cpp]: thats not a scalable approach

Comment (Gooyeth) [reply src/server/game/Spells/Auras/SpellAuras.cpp]: So what can be placed instead?

Comment (joschiwald) [reply src/server/game/Spells/Auras/SpellAuras.cpp]: maybe a custom spell attribute which is added on startup?

Comment (Gooyeth) [reply src/server/game/Spells/Auras/SpellAuras.cpp]: Ready, thanks

File: src/server/game/Spells/SpellMgr.cpp [hunk 2]
```diff
@@ -2946,6 +2946,15 @@ void SpellMgr::LoadSpellInfoCustomAttributes()
         }
     }
 
+    // add attribute custom to liquid auras
```
Comment (tkrokli) [inline src/server/game/Spells/SpellMgr.cpp]: Cosmetic/grammar: `// add custom attribute to liquid auras`

File: src/server/game/Spells/SpellMgr.cpp [hunk 3]
```diff
@@ -2946,6 +2946,12 @@ void SpellMgr::LoadSpellInfoCustomAttributes()
...
+    for (LiquidTypeEntry const* liquid : sLiquidTypeStore)
+        if (liquid->SpellId)
```
Comment (ariel-) [inline src/server/game/Spells/SpellMgr.cpp]: We use enclosing braces `{ }` following a for/if whenever the next instruction is a block

Comment (Gooyeth) [reply src/server/game/Spells/SpellMgr.cpp]: Ready, thanks

File: sql/updates/characters/3.3.5/9999_99_99_99_characters.sql [hunk 4]
```diff
@@ -0,0 +1,2 @@
+--
+DELETE FROM `pet_aura` WHERE `spell` IN (28801, 37025, 42201, 57634);
```
Comment (ariel-) [inline sql/updates/characters/3.3.5/9999_99_99_99_characters.sql]: One last thing before merge: the sql (and the characters_database.sql changes) can be dropped.

The core will automatically avoid loading any aura (and pet aura) which doesn't pass Aura::CanBeSaved

Comment (Gooyeth) [reply sql/updates/characters/3.3.5/9999_99_99_99_characters.sql]: Ready

**Comment by ariel-** [comment]
Welp, github isn't cooperating today.

Thanks for the contribution :)

**Pull request merged**